pkg/kubectl/cmd/top_pod.go

```diff
@@ -111,11 +119,68 @@ func (o *TopPodOptions) Validate() error {
 	return nil
 }
 
-// RunTop implements all the necessary functionality for top.
 func (o TopPodOptions) RunTopPod() error {
-	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, o.Selector)
+	var err error
+	selector := labels.Everything()
+	if len(o.Selector) > 0 {
+		selector, err = labels.Parse(o.Selector)
+		if err != nil {
+			return err
+		}
+	}
+	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, selector)
+	// TODO: Refactor this once Heapster becomes the API server.
```

Comment:
This is significant enough that I think we should have an issue for it.  Since you've done work in the area, it would be nice if you could own the issue until its fixed.